### PR TITLE
[DataFrame] Implemented nunique, skew

### PR DESCRIPTION
**Pull request by hsubbaraj**
Implemented nunique, skew

### CODE REVIEW

**Review by kunalgosar — CHANGES_REQUESTED** [review]
Looks really good! Just a few minor comments.

**Comment by hsubbaraj** [comment]
I took out the changes from #1994 as well

File: python/ray/dataframe/test/test_dataframe.py [hunk 1]
```diff
@@ -2205,14 +2203,14 @@ def test_pow():
     test_inter_df_math("pow", simple=False)
 
 
-def test_prod():
+def test_prod(ray_df, pandas_df):
```
Comment (kunalgosar) [inline python/ray/dataframe/test/test_dataframe.py]: These changes should be moved to #1994

File: python/ray/dataframe/test/test_dataframe.py [hunk 2]
```diff
@@ -2149,11 +2149,9 @@ def test_nsmallest():
...
 
 
-def test_nunique():
-    ray_df = create_test_dataframe()
-
-    with pytest.raises(NotImplementedError):
-        ray_df.nunique()
+@pytest.fixture
+def test_nunique(ray_df, pandas_df):
+    assert(ray_df_equals_pandas(ray_df.nunique(), pandas_df.nunique()))
```
Comment (kunalgosar) [inline python/ray/dataframe/test/test_dataframe.py]: Test both axis here.

**Review by kunalgosar — APPROVED** [review]
Few minor testing comments; generally, looks really good! Many of the early comments were addressed in #1994 after code was moved over.

**Review by kunalgosar — APPROVED** [review]
Looks great, thanks!

**Review by devin-petersohn — APPROVED** [review]
Looks great! Thanks @hsubbaraj!

**Pull request merged**